File: CustomerService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/customers';

class CustomerService {
  getCustomers() {
    return axios.get(API_URL);
  }

  addCustomer(customer) {
    return axios.post(API_URL, customer);
  }

  updateCustomer(id, customer) {
    return axios.put(`${API_URL}/${id}`, customer);
  }

  deleteCustomer(id) {
    return axios.delete(`${API_URL}/${id}`);
  }
}

export default new CustomerService();

File: CustomerService.java
package com.example.customer.service;

import com.example.customer.model.Customer;
import com.example.customer.repository.CustomerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class CustomerService {
    @Autowired
    private CustomerRepository repository;

    public List<Customer> getAllCustomers() {
        return repository.findAll();
    }

    public Customer addCustomer(Customer customer) {
        return repository.save(customer);
    }

    public Customer updateCustomer(Long id, Customer newCustomer) {
        return repository.findById(id).map(customer -> {
            customer.setName(newCustomer.getName());
            customer.setAddress(newCustomer.getAddress());
            customer.setPhone(newCustomer.getPhone());
            return repository.save(customer);
        }).orElseThrow(() -> new RuntimeException("Customer not found"));
    }

    public void deleteCustomer(Long id) {
        repository.deleteById(id);
    }
}
